pkg/virtctl/imageupload/imageupload.go

```diff
@@ -555,27 +557,11 @@ func createStorageSpec(client kubecli.KubevirtClient, size, storageClass, access
 		},
 	}
 
-	if storageClass == "" {
-		storageClass, err = getDefaultStorageClassName(client)
-		if err != nil {
-			return nil, err
-		}
-	}
-
 	if storageClass != "" {
 		spec.StorageClassName = &storageClass
 	}
 
-	if accessMode == "" {
-		accessModeAvailable, err := profileSpecifiesCorrectAccessMode(client, storageClass)
-		if err != nil {
-			return nil, err
-		}
-		if !accessModeAvailable {
-			// fallback to safe Default
-			spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}
-		}
-	} else {
+	if accessMode != "" {
```

Comment:
do we need to check the version of CDI deployed to ensure that not setting accessmode is allowed? Or is the expectation that the posting of the DataVolume will fail if accessmode is required in an older version of cdi?

Reply:
Thanks David, your question is actually a very good one. 

Here, we are using `storage` spec from CDI. And this is only available from https://github.com/kubevirt/containerized-data-importer/releases/tag/v1.34.0. So, it will not work for previous - older versions. It will not work even if user provides asccessMode. 

The virtctl is a client program, than can be downloaded and used for different cluster versions. I am not sure what is the policy about the versions supported. How many versions back? What to do to be good for users?

Reply:
> How many versions back? What to do to be good for users?

well... This is tricky because CDI versioning isn't tied to KubeVirt. Technically, we're only guaranteeing that virtctl will function with KubeVirt versions >= the version that virtctl is built from. That doesn't necessarily include CDI right now though. 

I think the expectation should be that people update CDI with KubeVirt, and given this was introduced all the way back in cdi v1.34, we should be fine in this case.

Reply:
@brybacki fyi, i altered the release note in the PR's description to make note of this.

Reply:
Thanks.